tools/exp/filter/filter.go

```diff
@@ -0,0 +1,147 @@
+package filter
+
+import (
+	"errors"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/mitchellh/copystructure"
+)
+
+// errUnexpected is the error that is returned when an unexpected error occurs.
+var errUnexpected = errors.New("unexpected filtering error")
+
+// supportedServiceTypes is a pseudo-constant map of supported service types.
+var supportedServiceTypes = func() map[string]struct{} {
+	return map[string]struct{}{
+		"alertmanager":                    {}, // As of v3.5.1 this is not supported in Aiven Terraform provider.
```

Comment:
`alertmanager` is an Aiven internal service, should not be here.

Reply:
I will rework the filter a little bit, same goes for all the below comments